protocol/xmrtaker/claim.go

```diff
@@ -0,0 +1,157 @@
+package xmrtaker
+
+import (
+	"fmt"
+
+	"github.com/athanorlabs/atomic-swap/common/types"
+	mcrypto "github.com/athanorlabs/atomic-swap/crypto/monero"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	"github.com/athanorlabs/atomic-swap/monero"
+
+	eth "github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+func (s *swapState) tryClaim() error {
+	if !s.info.Status.IsOngoing() {
```

Comment:
the functions in this file haven't been changed, i just moved them out of swap_state.go since that file was pretty large